src/EnergyPlus/DXCoils.cc

```diff
@@ -14655,140 +14655,111 @@ void ReportDXCoil(EnergyPlusData &state, int const DXCoilNum) // number of the c
-        state.dataDXCoils->DXCoil(DXCoilNum).LatCoolingEnergy =
-            state.dataDXCoils->DXCoil(DXCoilNum).TotalCoolingEnergy - state.dataDXCoils->DXCoil(DXCoilNum).SensCoolingEnergy;
-        state.dataDXCoils->DXCoil(DXCoilNum).ElecCoolingConsumption = state.dataDXCoils->DXCoil(DXCoilNum).ElecCoolingPower * ReportingConstant;
-        state.dataDXCoils->DXCoil(DXCoilNum).CrankcaseHeaterConsumption =
-            state.dataDXCoils->DXCoil(DXCoilNum).CrankcaseHeaterPower * ReportingConstant;
-        DXElecCoolingPower = state.dataDXCoils->DXCoil(DXCoilNum).ElecCoolingPower;
-        state.dataDXCoils->DXCoil(DXCoilNum).EvapCondPumpElecConsumption =
-            state.dataDXCoils->DXCoil(DXCoilNum).EvapCondPumpElecPower * ReportingConstant;
-        state.dataDXCoils->DXCoil(DXCoilNum).EvapWaterConsump = state.dataDXCoils->DXCoil(DXCoilNum).EvapWaterConsumpRate * ReportingConstant;
-        if (any_eq(state.dataDXCoils->DXCoil(DXCoilNum).CondenserType, DataHeatBalance::RefrigCondenserType::Evap)) {
-            state.dataDXCoils->DXCoil(DXCoilNum).BasinHeaterConsumption = state.dataDXCoils->DXCoil(DXCoilNum).BasinHeaterPower * ReportingConstant;
+        dxCoil.TotalCoolingEnergy = dxCoil.TotalCoolingEnergyRate * ReportingConstant;
+        dxCoil.SensCoolingEnergy = dxCoil.SensCoolingEnergyRate * ReportingConstant;
+        dxCoil.LatCoolingEnergy = dxCoil.TotalCoolingEnergy - dxCoil.SensCoolingEnergy;
+        dxCoil.ElecCoolingConsumption = dxCoil.ElecCoolingPower * ReportingConstant;
+        dxCoil.CrankcaseHeaterConsumption = dxCoil.CrankcaseHeaterPower * ReportingConstant;
+        state.dataHVACGlobal->DXElecCoolingPower = dxCoil.ElecCoolingPower;
+        dxCoil.EvapCondPumpElecConsumption = dxCoil.EvapCondPumpElecPower * ReportingConstant;
+        dxCoil.EvapWaterConsump = dxCoil.EvapWaterConsumpRate * ReportingConstant;
+        if (any_eq(dxCoil.CondenserType, DataHeatBalance::RefrigCondenserType::Evap)) {
+            dxCoil.BasinHeaterConsumption = dxCoil.BasinHeaterPower * ReportingConstant;
         }
     } break;
     }
 
-    if (state.dataDXCoils->DXCoil(DXCoilNum).CondensateCollectMode == CondensateCollectAction::ToTank) {
+    if (dxCoil.CondensateCollectMode == CondensateCollectAction::ToTank) {
         // calculate and report condensation rates  (how much water extracted from the air stream)
         // water flow of water in m3/s for water system interactions
         //  put here to catch all types of DX coils
-        Tavg = (state.dataDXCoils->DXCoil(DXCoilNum).InletAirTemp - state.dataDXCoils->DXCoil(DXCoilNum).OutletAirTemp) / 2.0;
-        RhoWater = RhoH2O(Tavg);
-        // CR9155 Remove specific humidity calculations
-        SpecHumIn = state.dataDXCoils->DXCoil(DXCoilNum).InletAirHumRat;
-        SpecHumOut = state.dataDXCoils->DXCoil(DXCoilNum).OutletAirHumRat;
+        Real64 Tavg = (dxCoil.InletAirTemp - dxCoil.OutletAirTemp) / 2.0;
```

Comment:
Noticed incorrect average temperature calculation, it should have been `Tavg = (dxCoil.InletAirTemp + dxCoil.OutletAirTemp) / 2.0`. Posted new issue [here](https://github.com/NREL/EnergyPlus/issues/9860).